### PR TITLE
Multi assets per vault

### PR DESCRIPTION
**Pull request by roienatan**
resolves: #137
resolves: #235 

- also includes `DepositeWitedraw.tsx` code improvements
- general code improvements for `vaults total prices in USD`, `severity rewards` and `vault apy` (see `useSeverityReward`, `useVaultsTotalPrices` and `useVaultsApy`)
- v1/v2 support (was merged to this PR)

- [x] Fix the case where the timer shows `-1:-1` for a brief moment before changing the state
- [x] Show the withdraw timer for a single token vault outside of the "deposit withdraw" modal
- [x] Merge `v1/v2 support` PR to this PR to avoid conflicts
- [x] Need to test vaults APY, handle the case of multi tokens vaults and add "%" sign.
- [x] Adjust **vault total** also in project select
- [x] Fixed assets table header
- [x] Change all new hardcoded text to support languages
- [x] Auto-select first option in the `TokenSelect`
- [x] Remove mock data from `useVaults`.
- [ ] Add shadow to the table header
- [ ] Active withdrawals on top of the table

### CODE REVIEW

File: src/components/Vault/hooks/useVaultsTotalPrices.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useVaults } from "hooks/useVaults";
+import { IVault } from "types/types";
+import { fromWei } from "utils";
+
+export function useVaultsTotalPrices(vaults: IVault[]) {
+  const { tokenPrices } = useVaults();
+
+  const totalPrices: { [token: string]: number } = {};
+
+  vaults.forEach(vault => {
```
Comment (leviyehonatan) [inline src/components/Vault/hooks/useVaultsTotalPrices.ts:10]: this case really fits reduce method

File: src/hooks/useVaults.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+export function VaultsProvider({ children }) {
+  const [vaults, setVaults] = useState<IVault[]>();
+  const [subscriptions, setSubscrptions] = useState<number>(0)
+  const prevSubscriptions = usePrevious(subscriptions);
+  const [tokenPrices, setTokenPrices] = useState<number[]>();
+  const apolloClient = useApolloClient();
+  const { chainId } = useEthers();
+  const prevChainId = usePrevious(chainId);
+
+  // useEffect(() => {
```
Comment (roienatan) [inline src/hooks/useVaults.tsx:42]: @leviyehonatan can we remove this block of comment?

File: src/components/DepositWithdraw/ApproveToken/ApproveToken.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
-import LockIcon from "../../assets/icons/lock.icon";
-import "../../styles/DepositWithdraw/ApproveToken.scss";
-import { toWei } from "../../utils";
 import ReactDOM from "react-dom";
+import LockIcon from "../../../assets/icons/lock.icon";
+import { toWei } from "../../../utils";
```
[the remark this answers is not in the document]
Comment (roienatan) [reply src/components/DepositWithdraw/ApproveToken/ApproveToken.tsx]: @leviyehonatan `fromWei` and `toWei` wrapper have been removed and now we use `ethers` functions directly.

File: src/hooks/useVaults.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+      return multiVault;
+    } else {
+      return vault;
+    }
+  })
+
+  return updatedVaults;
+}
+
+const fetchVaultsByPids = (vaults: IVault[], pids: string[]) => {
```
[the remark this answers is not in the document]
Comment (roienatan) [reply src/hooks/useVaults.tsx]: Done.

File: src/components/Vault/hooks/useVaultsTotalPrices.ts (hunk 1 above)
Comment (roienatan) [reply src/components/Vault/hooks/useVaultsTotalPrices.ts:10]: Mm not sure I understand. In that case we want to traverse the vaults and only if we are able to calculate `totalUSDValue` then we add it to our object. We have no accumulated value here.

File: src/hooks/useVaults.tsx (hunk 2 above)
Comment (leviyehonatan) [reply src/hooks/useVaults.tsx:42]: yes

File: src/components/Vault/hooks/useVaultsTotalPrices.ts (hunk 1 above)
Comment (leviyehonatan) [reply src/components/Vault/hooks/useVaultsTotalPrices.ts:10]: please refer to reduce example from Honeypots.tsx

File: src/components/DepositWithdraw/ApproveToken/ApproveToken.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (leviyehonatan) [reply src/components/DepositWithdraw/ApproveToken/ApproveToken.tsx]: thanks

**Review by leviyehonatan — COMMENTED** [review]
please refer to the comitted changes

**Review by leviyehonatan — DISMISSED** [review]

**Review by shayzluf — APPROVED** [review]
approving it for @leviyehonatan

**Pull request merged**